homeassistant/components/ws66i/config_flow.py

```diff
@@ -0,0 +1,149 @@
+"""Config flow for WS66i 6-Zone Amplifier integration."""
+import logging
+
+from pyws66i import get_ws66i
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_IP_ADDRESS
+
+from .const import (
+    CONF_SOURCE_1,
+    CONF_SOURCE_2,
+    CONF_SOURCE_3,
+    CONF_SOURCE_4,
+    CONF_SOURCE_5,
+    CONF_SOURCE_6,
+    CONF_SOURCES,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SOURCES = [
+    CONF_SOURCE_1,
+    CONF_SOURCE_2,
+    CONF_SOURCE_3,
+    CONF_SOURCE_4,
+    CONF_SOURCE_5,
+    CONF_SOURCE_6,
+]
+
+OPTIONS_FOR_DATA = {vol.Optional(source): str for source in SOURCES}
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_IP_ADDRESS): str, **OPTIONS_FOR_DATA})
+
+
+@core.callback
+def _sources_from_config(data):
+    sources_config = {
+        str(idx + 1): data.get(source) for idx, source in enumerate(SOURCES)
+    }
+
+    return {
+        index: name.strip()
+        for index, name in sources_config.items()
+        if (name is not None and name.strip() != "")
+    }
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    ws66i = get_ws66i(data[CONF_IP_ADDRESS])
+    try:
+        await hass.async_add_executor_job(ws66i.open)
+    except ConnectionError as err:
+        raise CannotConnect from err
```

Comment:
No need to create `CannotConnect` since the library already has an exception for this. You can use `ConnectionError` directly instead

Reply:
I removed the try block inside this method. It looks like my calling method catches the exception.